test-fixture/src/sim/mod.rs

```diff
@@ -151,23 +148,37 @@ impl Simulator {
             .into_iter()
             .chain(it.map(|node| NodeHolder { node, state: Idle }))
             .collect::<Vec<_>>();
-        Self {
+        let mut sim = Self {
             name,
             nodes,
             rng: Rc::default(),
+        };
+        // Seed from the `SIMULATION_SEED` environment variable, if set.
+        if let Ok(seed) = std::env::var("SIMULATION_SEED") {
+            sim.seed_str(seed);
         }
-    }
-
-    pub fn seed(&mut self, seed: [u8; 32]) {
-        self.rng = Rc::new(RefCell::new(Random::new(&seed)));
+        // Dump the seed to a file to the directory in the `DUMP_SIMULATION_SEEDS` environment
+        // variable, if set.
+        if let Ok(dir) = std::env::var("DUMP_SIMULATION_SEEDS") {
+            if create_dir_all(&dir).is_err() {
+                qerror!("Failed to create directory {dir}");
+            } else {
+                let seed_str = sim.rng.borrow().seed_str();
+                let path = format!("{dir}/{seed_str}");
+                if write(&path, seed_str).is_err() {
+                    qerror!("Failed to write seed to {path}");
```

Comment:
```suggestion
                    qerror!("Simulation seed: {seed_str}");
```
Might as well just print the value out here.

Reply:
It's being printed elsewhere already.